### PR TITLE
🎨 Better fetch session user data

### PR DESCRIPTION
**Pull request by nwingt**
image

### CODE REVIEW

**Review by AuroraHuang22 — APPROVED** [review]

File: src/store/modules/wallet.js [hunk 1]
```diff
@@ -230,17 +230,14 @@ const actions = {
     try {
       if (state.signer && !getters.walletIsMatchedSession) {
         await dispatch('signLogin');
+      } else if (getters.walletHasLoggedIn) {
```
Comment (williamchong) [inline src/store/modules/wallet.js:233]: The else relationship seems not intuitive, i.e. what does `!(state.signer && !getters.walletIsMatchedSession)` mean?
Maybe should name the if condition above to a meaningful name

Comment (nwingt) [reply src/store/modules/wallet.js:235]: Updated

Comment (williamchong) [reply src/store/modules/wallet.js:235]: Is it possible for `!state.signer & getters.walletIsMatchedSession` to exist?

Comment (nwingt) [reply src/store/modules/wallet.js]: Updated with flipped conditions

File: src/mixins/wallet.js [hunk 2]
```diff
@@ -72,7 +73,9 @@ export default {
         'connected_wallet',
         1
       );
-      return this.initWallet(connection);
+      return isSkippedLogin
```
Comment (williamchong) [inline src/mixins/wallet.js:76]: Does this flag means `Login was skipped` or it means we want to skip login when calling this function?
If it is latter, we should call this flag `skipLogin`/`isSkipLogin`

File: src/store/modules/wallet.js [hunk 3]
```diff
@@ -403,15 +410,28 @@ const actions = {
...
     try {
       const userInfo = await this.$api.$get(getUserV2Self());
-      commit(WALLET_SET_USER_INFO, userInfo || { user: address });
+      commit(WALLET_SET_USER_INFO, userInfo || { user: state.address });
       return userInfo;
     } catch (error) {
       throw error;
     }
   },
+  async walletFetchSessionUserData({ dispatch }, { isExtraOnly = false } = {}) {
```
Comment (williamchong) [inline src/store/modules/wallet.js:422]: `isSkipUserInfo` for consistent naming?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/mixins/wallet.js (hunk 2 above)
Comment (nwingt) [reply src/mixins/wallet.js:76]: How about `shouldSkipLogin`?

Comment (williamchong) [reply src/mixins/wallet.js:76]: Yes it would be nice

**Review by williamchong — APPROVED** [review]

**Pull request merged**